spec/classes/puppet_agent_install_spec.rb

```diff
@@ -8,7 +8,12 @@
     additional_facts = {:rubysitedir => '/opt/puppetlabs/puppet/lib/ruby/site_ruby/2.1.0'}
   end
 
-  let :common_centos_facts do on_supported_os['centos-6-x86_64'].merge({
+on_supported_os.each do |os, os_facts|
+  next if not os == "centos-6-x86_64"
```

Comment:
I'm wondering if we can generalize this. Maybe a variable or helper function in ```spec_helper``` to only test on the reduced set? The set can contain just 1 element for now, but I can see us testing this on CentOS latest and Debian latest to get coverage for both OS families.